sympy/physics/quantum/fermion.py

```diff
@@ -109,6 +109,23 @@ def _print_contents_pretty(self, printer, *args):
         else:
             return pform**prettyForm('\N{DAGGER}')
 
+    def _eval_power(self, exp):
+        from sympy.core.singleton import S
+        try:
+            if exp == 0:
+                return S.One
+            elif exp == 1:
+                return self
+            elif exp > 1 and exp.is_integer:
+                return S.Zero
+            else:
+                raise ValueError("Fermionic operators can only be raised to a"
+                    " positive integer power")
+        except TypeError as e:  # fallback to default implementation
```

Comment:
This happens only with `exp > 1`. Maybe can you just change to `(exp > 1) == True` not to catch the exceptions

Reply:
No, it also happens in the other clauses (`==` is also a relational)

Reply:
We implemented the conditions this way in order to allow exponentiation by a `Symbol` that could later be replaced. For example,
```py
c = FermionOp('c')
a = Symbol('a')

c**a
```
and then doing `c.subs({a: 2})` results in `0`.

Reply:
I don't think that using the try catch should be necessary for implementing that semantics. Especially, if you check with strings `e.args != ('cannot determine truth value of Relational',)`, there is chance that the message just changes and things stop working.

Reply:
That's true, I'll implement your suggestion, thanks!

Reply:
How would you go about `is_integer` checking? We can't negate it because `Symbol("a").is_integer == False`. Is there a way to ensure that something isn't an integer?

Reply:
`Symbol("a").is_integer` is `None`. SymPy uses [three-valued logic](https://docs.sympy.org/latest/guides/booleans.html#symbolic-boolean-vs-three-valued-bool) so checking syntactic equality with `False` is okay.